packages/web3-eth-abi/src/api/events_api.ts

```diff
@@ -0,0 +1,23 @@
+import { AbiError } from 'web3-common';
+import { sha3Raw } from 'web3-utils';
+import { JsonAbiEventFragment } from '../types';
+import { jsonInterfaceMethodToString, isAbiEventFragment } from '../utils';
+
+/**
+ * Encodes the function name to its ABI representation, which are the first 4 bytes of the sha3 of the function name including  types.
+ */
+export const encodeEventSignature = (functionName: string | JsonAbiEventFragment): string => {
+	if (typeof functionName !== 'string' && !isAbiEventFragment(functionName)) {
+		throw new AbiError('Invalid parameter value in encodeEventSignature');
+	}
+
+	let name: string;
+
+	if (typeof functionName === 'function' || (typeof functionName === 'object' && functionName)) {
```

Comment:
Technically existence of `functionName` should be at start of `encodeEventSignature` , but also if you are even checking it here its short circuit `&&` so good to first check if `functionName` exists like: 
```suggestion
	if (functionName && (typeof functionName === 'function' || typeof functionName === 'object')) {
```